JASP-Engine/JASP/R/anovarepeatedmeasures.R

```diff
@@ -2501,6 +2527,273 @@ AnovaRepeatedMeasures <- function(dataset=NULL, options, perform="run", callback
+    dfWithin <- fullAnovaTableWithin$data[[errorIndexWithin]]$df
+    dfBetween <- fullAnovaTableBetween$data[[errorIndexBetween]]$df
+    fullAnovaTable <- fullAnovaTableWithin
+    tableCounter <- 1
+    if (isMixedAnova) {
+      pooledSS <- ssWithin + ssBetween
+      pooledMS <- fullAnovaMS <- pooledSS / (dfWithin + dfBetween)
+      errorDfPooled <- fullAnovaDf <- pooledSS^2 / ( (ssWithin^2 / dfWithin) + (ssBetween^2 / dfBetween) )
+      if ( is.nan(errorDfPooled) ) {
+        .addFootnote(footnotes, text = "Error df and p-value undefined due to extreme values for SS", 
+                     symbol = "<em>Note.</em>")
+      }
+    } else {
+      fullAnovaMS <- fullAnovaTable$data[[length(fullAnovaTable$data)]]$MS 
+      fullAnovaDf <- fullAnovaTable$data[[length(fullAnovaTable$data)]]$df
+    }
+    
+  
+    simpleEffectRows <- list()
+    rows <- list()
+    
+    termsBothModerators <- as.vector(c(moderatorFactorOne, moderatorFactorTwo))
+    if(identical(moderatorFactorTwo, "")) {
+      termsBothModerators <- termsBothModerators[1]
+    }
+    
+    lvls <- list()
+    factors <- list()
+    
+    for (variable in termsBothModerators) {
+      if (variable %in% unlist(options$withinModelTerms)) {
+        whichFactor <- unlist(lapply(options$repeatedMeasuresFactors, 
+                                     FUN = function(x){x$name == variable}))
+        lvls[[variable]] <- options$repeatedMeasuresFactors[[which(whichFactor == TRUE)]]$levels
+      } else if (variable %in% unlist(options$betweenModelTerms)) {
+        factor <- dataset[[ .v(variable) ]]
+        factors[[length(factors)+1]] <- factor
+        lvls[[variable]] <- levels(factor)
+      }
+    }
+    #
+    allNames <- unlist(lapply(options$repeatedMeasuresFactors, function(x) x$name)) # Factornames 
+    
+    # Following steps to make sure column ordering is the same for the datasets
+    wideDataset <- fullModel$data$wide[,(-1)]
+    factorNamesV <- colnames(wideDataset)
+    withinFactorNamesV <- factorNamesV[!.unv(factorNamesV) %in% options$betweenSubjectFactors]
+
+    withinOrder <- match(wideDataset, dataset)
+    betweenOrder <- match(names(wideDataset), names(dataset))
+    betweenOrder[is.na(betweenOrder)] <- withinOrder[!is.na(withinOrder)]
+    fullOrder <- betweenOrder
+    
+    dataset <- dataset[fullOrder]
+    
+    orderOfTerms <- unlist(options$withinModelTerms[[length(options$withinModelTerms)]]$components)
+    indexofOrderFactors <- match(allNames,orderOfTerms)
+
+    for (level in lvls[[1]]) {
+      # For each level of the first moderator factor, take a subset of the dataset, and adjust the options object
+      # Suboptions is the same as options, except that the first moderator factor has been removed as a predictor 
+      # (because each subset only has one level of that factor). The same procedure is applied to the second moderator, if specified.
+      
+      subOptions <- options
+      if (isModeratorOneWithin) {
```

Comment:
Yeah alright this if/else block is very long. Do you think it would be shorter with a [`switch()`](https://stackoverflow.com/questions/10393508/how-to-use-the-switch-statement-in-r-functions) statement? If not, leave it like this but maybe add some comments within each condition to show which case we're in.